src/plugins/data/server/search/search_service.ts

```diff
@@ -315,6 +315,19 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
     return strategy.cancel(id, options, deps);
   };
 
+  private extend = (
+    id: string,
+    keepAlive: string,
+    options: ISearchOptions,
+    deps: SearchStrategyDependencies
+  ) => {
+    const strategy = this.getSearchStrategy(options.strategy);
+    if (!strategy.extend) {
+      throw new Error(`Search strategy ${options.strategy} does not support extend`);
+    }
+    return strategy.extend(id, keepAlive, options, deps);
```

Comment:
Thoughts about validating the `keepAlive` string? It looks like it should be a valid time units string: https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#time-units

Reply:
After giving this some thought, I think we can rely on the error/message we get back from Elasticsearch for this:

```json
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "failed to parse setting [keep_alive] with value [foobar] as a time value: unit is missing or unrecognized"
            }
        ],
        "type": "illegal_argument_exception",
        "reason": "failed to parse setting [keep_alive] with value [foobar] as a time value: unit is missing or unrecognized"
    },
    "status": 400
}
```